src/index.js

```diff
@@ -311,63 +311,106 @@ class EventsCoreAPI {
    * Create a webhook or journal registration
    *
    * @param {string} consumerOrgId Consumer Org Id from the console
-   * @param {string} integrationId integration Id from the console
+   * @param {string} projectId Project Id from the console
+   * @param {string} workspaceId Workspace Id from the console
    * @param {object} body Json data contains details of the registration
    * @returns {Promise<object>} Details of the webhook/journal registration created
    */
-  createWebhookRegistration (consumerOrgId, integrationId, body) {
+  createRegistration (consumerOrgId, projectId, workspaceId, body) {
     const headers = {}
     const requestOptions = this.__createRequest('POST', headers, JSON.stringify(body))
-    const url = this.__getUrl(`/events/organizations/${consumerOrgId}/integrations/${integrationId}/registrations`)
+    const url = this.__getUrl(`/events/${consumerOrgId}/${projectId}/${workspaceId}/registrations`)
+    const sdkDetails = { requestOptions: requestOptions, url: url }
+    return this.__handleRequest(sdkDetails, codes.ERROR_CREATE_REGISTRATION)
+  }
+
+  /**
+   * Update a webhook or journal registration
+   *
+   * @param {string} consumerOrgId Consumer Org Id from the console
+   * @param {string} projectId Project Id from the console
+   * @param {string} workspaceId Workspace Id from the console
+   * @param {string} registrationId Registration id whose details are to be fetched
+   * @param {object} body Json data contains details of the registration
+   * @returns {Promise<object>} Details of the webhook/journal registration to be updated
+   */
+  updateRegistration (consumerOrgId, projectId, workspaceId, registrationId, body) {
+    const headers = {}
```

Comment:
Why do we need this empty `headers` object?

Reply:
the __createRequest() method is expecting a headers map with any additional headers that may be required in addition to the default ones. in this case, only the default headers apply and hence we are sending an empty headers object.